Remove surrounding backticks before parsing output as JSON

irgolic opened the pull request:
`gpt-3.5-turbo` consistently returns the output surrounded by backticks for me:

E.g.,
```
```json
{
    "action": "new_file",
    "new_file": {
        "filepath": "tic_tac_toe.py",
        "description": "Python file containing the implementation of the Tic Tac Toe game and AI"
    }
}
\```
```
(I added the \ on the closing backticks to avoid closing the codeblock)

This PR removes the backticks before JSON parsing, should they appear.

irgolic committed:
run: Remove surrounding backticks before parsing as JSON

ShreyaR reviewed (commented):
Looks good!

I can approve once tests are passing.

Btw, have you tried out the prompting technique where you most of the prompt in the system messages, and only pass in the low level task details in the user message? Curious if you're still getting backticks with that approach.

irgolic committed:
llm_output: Remove preceding \n

irgolic commented:
Fixed tests, there seems to be some weird behavior that's different on 3.8 and 3.9 in the SQL test I fixed.

> Btw, have you tried out the prompting technique where you most of the prompt in the system messages, and only pass in the low level task details in the user message? Curious if you're still getting backticks with that approach.

I've not! Is there an easy way to transition to this system? :D
